apps/ledger-live-mobile/src/screens/BleDevicePairingFlow/index.tsx

```diff
@@ -134,46 +133,8 @@ export const BleDevicePairingFlow = ({ navigation, route }: Props) => {
     [navigateInput, navigation, navigationType, pathToDeviceParam],
   );
 
-  const handleGoBackFromScanning = useCallback(() => {
-    const routes = navigation.getState().routes;
-
-    const isNavigationFromDeeplink =
-      routes[routes.length - 1]?.params === undefined;
-
-    if (!isNavigationFromDeeplink) {
-      navigation.goBack();
-    } else {
-      navigation.reset({
-        index: 0,
-        routes: [
-          {
-            // @ts-expect-error is fixed in a future screen
-            name: NavigatorName.BaseOnboarding,
-            state: {
-              routes: [
-                {
-                  name: ScreenName.OnboardingWelcome,
-                },
-              ],
-            },
-          },
-        ],
-      });
-    }
-  }, [navigation]);
-
-  // Handles back button, necessary when the user comes from the deep link
-  useEffect(() => {
-    const listener = BackHandler.addEventListener("hardwareBackPress", () => {
-      handleGoBackFromScanning();
-      return true;
-    });
-
-    return () => listener.remove();
-  }, [handleGoBackFromScanning]);
-
```

Comment:
This workaround is no longer needed due to the change of `initialRouteName` in `DeeplinksProvider.tsx`.

Reply:
Nice !